src/types/contract_source_metadata/validate.rs

```diff
@@ -120,21 +118,6 @@ impl super::build_info::BuildInfo {
         Ok(())
     }
 
-    pub fn validate_build_command_on_whitelist(&self, entry: WhitelistEntry) -> eyre::Result<()> {
-        let expected_cmd_len = entry.expected_command_prefix.len();
-        if (self.build_command.len() < expected_cmd_len)
-            || (self.build_command[1..expected_cmd_len]
-                != entry.expected_command_prefix[1..expected_cmd_len])
-        {
-            return Err(eyre::eyre!(
-                "build_command {:?} must start with expected whitelist command prefix {:?}",
-                self.build_command,
-                entry.expected_command_prefix
-            ));
-        }
-        Ok(())
-    }
-
```

Comment:
it has been figured out that simple build command validation isn't enough in general case to ensure build correctness. 
For audit full source code must be inspected to ensure expected behaviour, starting from the build command as the entry point. 

One could try to do funky/intentionally wasteful things in e.g. build.rs of a contract, e.g. downloading a funky wasm from elsewhere and presenting it as the result of the build of a subcontract instead of building it from source, which would be apparently present in repo.